src/main/drivers/mcu/stm32/serial_uart_stm32f4xx.c

```diff
@@ -337,9 +337,7 @@ uartPort_t *serialUART(UARTDevice_e device, uint32_t baudRate, portMode_e mode,
 
     if (options & SERIAL_BIDIR) {
         IOInit(txIO, OWNER_SERIAL_TX, RESOURCE_INDEX(device));
-        IOConfigGPIOAF(txIO, (options & SERIAL_BIDIR_PP_PD) ? IOCFG_AF_PP_PD 
-                             : (options & SERIAL_BIDIR_PP) ? IOCFG_AF_PP
-                             : IOCFG_AF_OD, hardware->af);
+        IOConfigGPIOAF(txIO, ((options & SERIAL_BIDIR_PP) || (options & SERIAL_BIDIR_PP_PD)) ? IOCFG_AF_PP : IOCFG_AF_OD, hardware->af);
```

Comment:
OK, this changes GPIO_PuPd_DOWN to  GPIO_PuPd_NOPULL.

If this does fix the issue, it would probably mean that there is pullup on Smartaudio pin. That is against SA specification.

And disabling pulldown can cause problems with correct SA implementations.

Reply:
Also, at least G4 won't  work with that SA implementation

Reply:
@bkleiner did you test F4 / G4 with SA ?

Reply:
TBH as I was only rebasing and resubmitting @freasy changes, I had only given them a quick try on at32.